src/lib/y2firewall/clients/auto.rb

```diff
@@ -79,16 +81,18 @@ def summary
       #
       # @param profile [Hash] firewall profile section to be imported
       # @return [Boolean]
-      def import(profile, merge = !Yast::Mode.config)
+      def import(profile, merge = !Yast::Mode.config || !Yast::Mode.auto)
         self.class.profile = profile
         return false if merge && !read(force: false)
 
-        # Obtains the default from the control file (settings) if not present.
-        enable if profile.fetch("enable_firewall", settings.enable_firewall)
-        start if profile.fetch("start_firewall", false)
-        autoyast.import(profile)
-        check_profile_for_errors
-        imported
+        if Yast::Stage.cont
```

Comment:
I would move this responsibility into the autoinstallation module. In fact, one of the goals was to remove the section once imported.

https://github.com/yast/yast-autoinstallation/pull/653#discussion_r457214395

At that point we know exactly if it should be configured during the first stage or not. One of the main problems is that we need the package already installed to read the default configuration and merge with the one from the profile.

Reply:
it was and is my big dilemma. Putting it directly into AY is not problem. But on the other hand I have a feeling that it is also responsibility of the firewall module. And I'm still unsure ;-)

I'll prepare also a version with logic moved to AY, possibly it will add some points and break 50 : 50 problem ;-)